content/tutorials/Straightforward-Video-calls-with-WebRTC-and-Ably.textile

```diff
@@ -0,0 +1,408 @@
+        var video = document.getElementById('remote')
+        video.src = window.URL.createObjectURL(stream)
+        video.play()
+        renderMembers()
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/f56a55b551ba06c6118ab7e76f3bf1b8c3e9c108
+
+Here, we defined the @initiateCall@ function, which gets the stream from the user via @navigator.mediaDevices.getUserMedia@, then passes it to the @Connection@ class, to initiate a connection to the other peer.
+
+If you remember, in step 4, when we defined the @Connection@ class, we handled the @_onSignal@ event using @Ably@. Here, we defined the @rtc-signal/${clientId}@ event for that purpose.
+
+Since we are attempting to connect to the other peer, this event would be fired. So in this event, we check if a stream is assigned, if not, get the user stream, and pass it to the connect function. If assigned, call the @connect@ function
+
+In the connect function, we check if the connection already exists. If it dosen’t, create a new connection instance with the stream passed into it, else if it does exist, send the signal to the connection.
+
+Next, you would see the @recieveStream@ function. This function was called in step 4, under the @_onStream@. What it does, is to load the remote stream into the second video tag in our markup. 
+
+That’s it, we are connected to the other peer, and we can see a live video.
+
+How do we end calls then?
+
+let’s take a look at the function below:
+
+```[javascript]
+    function handleEndCall(client_id = null) {
+        if (client_id && client_id != currentCall) {
+            return
+        }
+        client_id = currentCall;
+        alert('call ended')
+        currentCall = undefined
+        connections[client_id].destroy()
+        delete connections[client_id]
+        for (var track of localStream.getTracks()) {
+            track.stop()
+        }
+        localStream = undefined
+        document.getElementById('call').style.display = 'none'
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/2c692c8d72bb0acdce4af83907767116c81d6b29
+
+Looking at the function above, we notice that we receive an optional argument of @client_id@. Why is this argument optional?
+
+Remember in the @Connection@ class, the @_onClose()@ method calls this function, passing in the client id of the other peer whose connection was closed/dropped. We use this parameter, to know which connection to delete from our @connections@ object. This function is also called when you click the @end call@  button, but here, no parameter is passed, as we have a variable called @currentCall@, which holds the value of the call you must be ending. 
+
+In this function, we destroy the peer connection, as well as deleting the key from our connection objects. We go ahead to stop the Video and Audio streams coming from the webcam, then hide the video inputs.
+
+h2.
+  default: Step 7 - Conclusion
```

Comment:
Please add testing/implementation steps in detail. Also please add an embedded demo as mentioned in the previous lesson.